samcli/commands/build/click_container.py

```diff
@@ -0,0 +1,27 @@
+"""
+Module to check container based cli parameters
+"""
+import click
+
+
+class ContainerOptions(click.Option):
+    """
+    Preprocessing checks for presence of --use-container flag for container based options.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.require_container: bool = kwargs.pop("require_container", False)
+
+        super().__init__(*args, **kwargs)
+
+    def handle_parse_result(self, ctx, opts, args):
+        if self.require_container:
+            if "use_container" not in opts and opts.get(self.name) is not None:
```

Comment:
we are using two forms of "check if X is in opts"
- `X in opts`
- `opts.get(X) is not None`

Is there a reason?

Reply:
There is a possibility that it is in opts but it is None, that's what our mock test does. In practice this doesn't occur. But they do check different things

Reply:
If in practice it never occurs, our unit test should not test it that way then. This inconsistency makes me confused. I am not sure if it is within the scope of this PR, if will be nice if we can unify them